### PR TITLE
ci: upgrade localstack to v3.3.0

### PR DESCRIPTION
**Pull request by rchincha**
There are performance improvements in recent releases of localstack.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -42,8 +42,8 @@ jobs:
       - name: Install localstack
         run: |
           pip install --upgrade pyopenssl
-          pip install localstack==2.3.1 awscli-local[ver1] # install LocalStack cli and awslocal
-          docker pull ghcr.io/project-zot/ci-images/localstack:2.3.1 # Make sure to pull the latest version of the image
+          pip install localstack==3.3.0 awscli-local[ver1] # install LocalStack cli and awslocal
```
Comment (andaaron) [inline .github/workflows/nightly.yaml:45]: Why does this .yaml not install python '3.11' and the others are?

File: .github/workflows/ecosystem-tools.yaml [hunk 2]
```diff
@@ -61,10 +61,13 @@ jobs:
...
           password: ${{ github.token }}
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
       - name: Install localstack
         run: |
           pip install --upgrade pyopenssl
-          pip install localstack==2.3.1 awscli-local[ver1] # install LocalStack cli and awslocal
+          pip install localstack==3.3.0 awscli-local[ver1] # install LocalStack cli and awslocal
           docker pull ghcr.io/project-zot/ci-images/localstack:2.3.1  # Make sure to pull a working version of the image
```
Comment (andaaron) [inline .github/workflows/ecosystem-tools.yaml:71]:
```suggestion
          docker pull ghcr.io/project-zot/ci-images/localstack:3.3.0  # Make sure to pull a working version of the image
```
?

File: .github/actions/setup-localstack/action.yaml [hunk 3]
```diff
@@ -18,11 +18,14 @@ runs:
...
         username: ${{ inputs.username }}
         password: ${{ inputs.password }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
     - shell: bash
       run: |
-            pip install localstack==2.3.1                                # Install LocalStack cli
+            pip install localstack==3.3.0                                # Install LocalStack cli
             # Below image was copied manually from localstack/localstack:2.2 and uploaded to ghcr
```
Comment (andaaron) [inline .github/actions/setup-localstack/action.yaml:27]: Update the comment to reflect current process.

File: .github/workflows/ecosystem-tools.yaml (hunk 2 above)
Comment (rchincha) [reply .github/workflows/ecosystem-tools.yaml:71]: What is the concern here?

Comment (andaaron) [reply .github/workflows/ecosystem-tools.yaml:71]: You are updating the client but not the server?
There is a version mismatch between them.

Comment (rchincha) [reply .github/workflows/ecosystem-tools.yaml]: Fixed.

File: .github/actions/setup-localstack/action.yaml (hunk 3 above)
Comment (rchincha) [reply .github/actions/setup-localstack/action.yaml:27]: Fixed.

Comment (andaaron) [reply .github/actions/setup-localstack/action.yaml:27]: Did you push this change?

**Review by andaaron — APPROVED** [review]

**Pull request merged**